temperature/routers.py

```
@@ -0,0 +1,40 @@
+from datetime import datetime
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from temperature import crud, schemas
+from database import AsyncSessionLocal
+
+router_temp = APIRouter()
+
+
+async def get_db():
+    async with AsyncSessionLocal() as db:
+        yield db
+
+
+@router_temp.get("/", response_model=list[schemas.Temperature])
+async def read_temp(db: AsyncSession = Depends(get_db)):
+    return await crud.get_temp(db=db)
+
+
+@router_temp.get("/{city_id}/", response_model=schemas.Temperature)
+async def read_single_temp(city_id: int, db: AsyncSession = Depends(get_db)):
+    temp = await crud.get_single_temp(db=db, city_id=city_id)
+    if temp is None:
```

The `crud.get_single_temp` function should return a temperature object or raise an exception if not found. Ensure that the CRUD function is correctly implemented to handle this.